paimon-common/src/main/java/org/apache/paimon/data/columnar/heap/AbstractStructVector.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.data.columnar.heap;
+
+import org.apache.paimon.data.columnar.ColumnVector;
+import org.apache.paimon.data.columnar.writable.WritableColumnVector;
+
+import java.util.Arrays;
+
+/** * Abstract class for vectors that have children. */
+public abstract class AbstractStructVector extends AbstractHeapVector
+        implements WritableColumnVector {
+
+    protected ColumnVector[] children;
+    protected long[] offsets;
```

Comment:
We should introduce a `AbstractArrayBasedVector` for array and map to contain offsets and lenths..